core/src/health.rs

```diff
@@ -0,0 +1,214 @@
+        services: HealthServices {
+            database: "unknown".to_string(),
+            indexing: "unknown".to_string(),
+            sync: "unknown".to_string(),
+        },
+        indexing: IndexingStatus {
+            active_tasks: active_indexing_count(),
+            is_running: is_running(),
+        },
+    };
+
+    // Check database connection if PostgreSQL is enabled
+    if state.manifest.storage.postgres_enabled() {
+        match &state.postgres_client {
+            Some(client) => {
+                match client.query_one("SELECT 1", &[]).await {
+                    Ok(_) => {
+                        health_status.services.database = "healthy".to_string();
+                    }
+                    Err(e) => {
+                        health_status.services.database = "unhealthy".to_string();
+                        health_status.status = "unhealthy".to_string();
+                        error!("Database health check failed: {}", e);
+                    }
+                }
+            }
+            None => {
+                health_status.services.database = "not_configured".to_string();
+                health_status.status = "unhealthy".to_string();
+            }
+        }
+    } else {
+        health_status.services.database = "disabled".to_string();
+    }
+
+    // Check indexing status
+    if health_status.indexing.is_running {
+        health_status.services.indexing = "healthy".to_string();
+    } else {
+        health_status.services.indexing = "stopped".to_string();
+        health_status.status = "unhealthy".to_string();
+    }
+
+    // Check sync status (basic check for event tables if using PostgreSQL)
+    if state.manifest.storage.postgres_enabled() {
+        match &state.postgres_client {
+            Some(client) => {
+                match client.query_one(
+                    "SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = 'public' AND table_name LIKE '%_events'",
+                    &[]
+                ).await {
+                    Ok(row) => {
+                        let count: i64 = row.get(0);
+                        if count > 0 {
+                            health_status.services.sync = "healthy".to_string();
+                        } else {
+                            health_status.services.sync = "no_data".to_string();
+                        }
+                    }
+                    Err(e) => {
+                        health_status.services.sync = "unhealthy".to_string();
+                        health_status.status = "unhealthy".to_string();
+                        error!("Sync health check failed: {}", e);
+                    }
+                }
+            }
+            None => {
+                health_status.services.sync = "not_configured".to_string();
+            }
+        }
+    } else {
+        // For CSV storage, check if the output directory exists and has files
+        if state.manifest.storage.csv_enabled() {
```

Comment:
method to nested shouldnt need to write it like this once you split this function up and you see the after you understand what i mean